.github/workflows/build-artifacts.yml

```diff
@@ -0,0 +1,64 @@
+name: Build GFM artifacts
+
+# Builds the GFM library as artifacts that can be used by other workflows
+#
+# It checks for any existing artifacts with an expiry date at least 10 days after today
+# If none are found, the library is built and uploaded.
+
+# This workflow can be run on demand, but is intended to be run on a weekly schedule
+# This should ensure that there is always at least one artifact available but no more than 2
+
+on:
+  # Schedule once a week
+  schedule:
+    - cron: '11 12 * * 0'
+  push:
+    paths:
+      - '**/build-artifacts.yml' # self
+      - 'pelican/build-cmark.sh'
+  workflow_dispatch:
+
+jobs:
+  build-artifact:
+    runs-on: ubuntu-latest
+    strategy:
+      # Allow for multiple versions to be maintained
+      matrix:
+        gfm_version: ['0.28.3.gfm.12']
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Fetch and build version ${{ matrix.gfm_version }}
+      id: build_gfm
+      run: |
+        # N.B. This must agree with the definition in pelican/action.yml
+        export GFM_ARTIFACT_KEY=gfm-lib-${{ matrix.gfm_version }}
+        echo "GFM_ARTIFACT_KEY=${GFM_ARTIFACT_KEY}" >> $GITHUB_ENV
+
+        # Check if artifact is present (list all)
+        curl -sS https://api.github.com/repos/sebbASF/infra-actions/actions/artifacts?name=${GFM_ARTIFACT_KEY} >/tmp/artifact.json
+        # when does last one expire?
+        jq </tmp/artifact.json '[.artifacts[]|select(.expired==false)|.expires_at]|max' >/tmp/max.txt
+        # is that more than 10 days away? (86400*10 seconds)
+        OUT=$(jq </tmp/max.txt 'select(.>(now+864000|strftime("%FT%TZ")))')
+        if [[ -n $OUT ]]
+        then
+          echo "Found a valid artifact for ${{ matrix.gfm_version }} (expires $OUT)"
```

Comment:
you should never use `${{ ... }}` inside a `run:` block.

use:
```yaml
env:
  gfm_version: ${{ matrix.gfm_version }}
...
run:
          echo "Found a valid artifact for $gfm_version (expires $OUT)"
...
```

Reply:
Depending on what you're doing, it can result in security vulnerabilities or very surprising misbehaviors.

In this case, if your matrix was:
```yaml
matrix:
  gfm_version:
  - 'hello"; touch /tmp/flag; echo "'
```

then you'd be constructing a script of the form:
```sh
          echo "Found a valid artifact for hello"; touch /tmp/flag; echo " (expires $OUT)"
```

which is fairly surprising.
https://github.com/check-spelling-sandbox/sturdy-adventure/actions/runs/13166687061/job/36748420310#step:6:14